.ci/linux-steps.yaml

```diff
@@ -72,14 +76,18 @@ steps:
     cmake $CMAKEARGS -DPYTHON_EXECUTABLE=`which python` -DCEREAL_INCLUDE_DIR=/usr/include/ ..
   displayName: 'CMake'
 
-# Build mlpack
-- script: cd build && make -j2
+# Build mlpack (TODO: maybe remove VERBOSE=1)
+- script: cd build && VERBOSE=1 make -j2 
   displayName: 'Build'
 
 # Run tests via ctest.
 - script: cd build && CTEST_OUTPUT_ON_FAILURE=1 ctest -T Test .
   displayName: 'Run tests via ctest'
 
+# Ccache stats (two verbosity levels supported
+- script: ccache --show-stats --verbose --verbose
+  displayName: 'Show ccache stats'
```

Comment:
Nice to have this at a quick glance.  I am not an expert with ccache, but it seems like it may be having fewer hits than I might have expected?

```
  /usr/bin/ccache -s
  Summary:
    Hits:               1 /  145 (0.69 %)
      Direct:           1 /  145 (0.69 %)
      Preprocessed:     0 /  144 (0.00 %)
    Misses:           144
      Direct:         144
      Preprocessed:   144
    Uncacheable:      175
  Primary storage:
    Hits:             146 /  290 (50.34 %)
    Misses:           144
    Cache size (GB): 0.40 / 0.50 (80.64 %)
    Cleanups:          11
```

I guess that maybe we can expect better performance once we actually deploy this, so it can benefit from previous builds in the cache?

Reply:
I do not know. I have the same gut feel as you here. 

I have used `ccache` _locally_ for many years, but had not tried it at GHA so I do not have much experience with cache persistence and availability.  What I did to test this in the scrap repo was to 'run once, then force a re-run via `workflow_dispatch`' and that way we _know_ a cache is useable and used.  That works and at least "proves" the setup works:

![image](https://github.com/user-attachments/assets/a0905a08-8453-4dec-8d48-e4f7bf944bac)

I see the PR as dipping a toe in the water.  Even 70 is a decent amount less than 100, and we can learn more.  Maybe, just maybe, we need to persist caches under our control---and `sccache` offers that but then we need to pay for storage somewhere.  Tradeoffs, tradeoffs.

Reply:
Agree, and maybe I didn't really look at the right builds to see an improvement.  Let's merge it, and maybe we can find more improvements along the way in the future. :+1: